### DISCUSSION TITLE
How to Trace Framework or Library Source Code Execution through VizTracer?

### DISCUSSION THREAD

**Discussion by AnElegantHusky** [gaogaotiantian/viztracer#483] [Q&A]
I am currently using VizTracer to trace the execution of a framework or library's source code. I attempted to include the source files by using the `--include_files` option, but I only managed to capture `builtins.exec.`

Does anyone have experience with tracing the execution of specific library or framework source code using VizTracer? Is there a different approach I should be using?

Any guidance or suggestions would be greatly appreciated. Thank you!

**Comment by gaogaotiantian** [comment]
The problem is - viztracer records function stack, you can't simply ignore an arbitrary piece of the stack and record the rest - the call relation would be messed up. So for viztracer, as long as you choose to ignore one function, you would ignore all it's sub-calls. I wouldn't recommend recording only code in a framework because normally that does not make sense - what if it calls to the standard library? Will the call be recorded? If the disk and the memory permits, it's better to record everything.

If you do need to only record piece of code, you can try `log_sparse` feature. Like I mentioned above, it's impossible (and not useful) to take out the middle calls and leave the rest. The best you can do is only record some calls and their sub-calls.

**Reply by AnElegantHusky** [reply]
Thank you so much for your prompt reply! I really appreciate your detailed explanation, and now I fully understand how to proceed based on your suggestion.

Also, VizTracer has been very helpful for my work—great tool!

**Reply by gz1z** [reply]
@gaogaotiantian , I almost agree, however, what if I'm interested the under the hood of a library and not want to get noise from asyncio or others. Thanks!

**Reply by gaogaotiantian** [reply]
There are a few ways to do it. You can use `log_sparse`(https://viztracer.readthedocs.io/en/latest/filter.html#log-sparse) if you have any control of the source code. Or you can manually filter the output json file based on the file name (it's just chrome trace events). However, for me, most of the cases, I just ignore the "noise" and focus on what I need. I don't think having extra information in the report bothers me too much.

**Comment by gz1z** [comment]
@AnElegantHusky , did you manage to figure this one out? I'm also interested in just tracing a library.
Thanks!
